### PR TITLE
add more postgres mock

### PR DESCRIPTION
**Pull request by PooyaRaki**
## Summary
This PR introduces additional mocks to simulate PostgreSQL interactions more flexibly in our Jest test suite

## Changes
entity-manager.mock.ts
postgresql-database.service.mock.ts
postgresql-datasource.mock.ts
querybuilder.mock.ts
repository.mock.ts

### CODE REVIEW

File: src/datasources/db/v2/__tests__/querybuilder.mock.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const mockQueryBuilder = {
+  delete: jest.fn().mockReturnThis(),
+  from: jest.fn().mockReturnThis(),
+  where: jest.fn().mockReturnThis(),
+  execute: jest.fn(),
+};
```
Comment (iamacook) [inline src/datasources/db/v2/__tests__/querybuilder.mock.ts:6]: Nit: let's type this to keep maintain type safety.

File: src/datasources/db/v2/database-migrator.service.spec.ts [hunk 2]
```diff
@@ -68,7 +66,7 @@ describe('PostgresDatabaseService', () => {
       providers: [DatabaseMigrator],
     }).compile();
 
-    connection = postgresDataSourceMock;
+    connection = mockPostgresDataSource;
```
Comment (iamacook) [inline src/datasources/db/v2/database-migrator.service.spec.ts:69]: This can be assigned when the variable is declared.

Comment (PooyaRaki) [reply src/datasources/db/v2/database-migrator.service.spec.ts]: Done https://github.com/safe-global/safe-client-gateway/pull/2084/commits/d879b9176acbbdeeb93f2a51942c5afc52149ec4

File: src/datasources/db/v2/__tests__/querybuilder.mock.ts (hunk 1 above)
Comment (PooyaRaki) [reply src/datasources/db/v2/__tests__/querybuilder.mock.ts]: Done https://github.com/safe-global/safe-client-gateway/pull/2084/commits/d879b9176acbbdeeb93f2a51942c5afc52149ec4

**Review by hectorgomezv — APPROVED** [review]
👏🏻

**Review by iamacook — APPROVED** [review]

**Pull request merged**